### ISSUE TITLE
Contributing ERPNext Quickbooks Connector to ERPNext.

### ISSUE THREAD

**Issue by jsukrut** [frappe/erpnext#15304]
**Contributing ERPNext Quickbooks Connector to ERPNext.**

ERPNext Quickbooks Connector is developed and enhanced by Indictrans with the help and support of community and Parsimony. With its intention to be contributed to core attempted earlier but left out to be an app, re-initiating contribution to the core as an enhanced version supporting oAuth2 and updated version of both ERPNext and QuickBooks.

ERPNext Quickbooks Connector is based on the integration of QuickBooks and ERPNext where QuickBooks Customers, Suppliers, Employees, Items, Sales Invoice, Purchase Invoice, Terms and Conditions, Journal Entry and Accounts are synchronized with ERPNext.

For the above contribution, we are not contributing it in bulk in support of a policy to have it in smaller PRs. We are starting with the first PR and add further in series.

Following are features:
1. Quickbooks Setting(Configuration) page.
2. Record Level Syncing
	1. Customers
	2. Suppliers
	3. Employees
	4. Items
	5. Sales Invoice
	6. Purchase Invoice
	7. Terms and Conditions
	8. Journal Entry 
	9. Accounts
3. Syncing will be done by the button to call the sync method manually.
4. Syncing Process will be displayed by Process Bar.
5. Documentation for understanding connection and syncing process

**Comment by adityahase** [comment]
@jsukrut Is this basically same as this [erpnext_quickbooks app](https://github.com/indictranstech/erpnext_quickbooks)

At Frappe, I have been assigned with the Quickbooks Migrator project.

The erpnext_quickbooks app gave us really good ideas and it have been a significant help in writing the new QB Migrator, However, We have taken some different approaches than the ones have been taken in the erpnext_quickbooks app. The new Migrator also fetches certain type of records that weren't already fetched by the erpnext_quickbooks app.

We already have spent a significant amount of resources in writing the Quickbooks Migrator, It'll be ready to be merged in a couple of weeks.

**Comment by jsukrut** [comment]
As ERPNext Quickbooks Connector is not the only migrator we used to sync dual way i.e Quickbooks to ERPNext and ERPNext QuickBooks.

- Sales / Purchase Invoice will be created in ERPNext and synced with Quickbooks.
- Customers, Suppliers, Item, Tax, Employee, and User will be managed through ERPNext and 
  whenever added new or updated existing one, it will update in Quickbooks.
- Journal Entry against SI/PI will be primarily created in Quickbooks and same will be synced to 
  ERPNext or vice versa.
- Migrate opening data from Quickbooks to ERPNext.

It is quite surprising to us as we don't know about the same and from our end, we also spent a lot of efforts for doing this.
